spec/index.bs

```diff
@@ -13185,6 +13314,8 @@ interface GPUCanvasContext {
 
     **Returns:** image contents
 
+    [=Content timeline=] steps:
+
     1. Ensure that all submitted work items (e.g. queue submissions) have
         completed writing to the image (via |context|.{{GPUCanvasContext/[[currentTexture]]}}).
     1. Let |snapshot| be a copy of |context|.{{GPUCanvasContext/[[drawingBuffer]]}}.
```

Comment:
aside: Noticed we'll probably want to sort out some timeline shenanigans here because this is the one place where we may need to synchronously wait for a result back from the device timeline - technically only for readbacks visible to JS but I think we are supposed to pretend the compositor etc run on the main thread too. (I had a brief thought that maybe we could pretend all our stuff runs on the main thread too, which would be more web-spec-like, now that we have this nice timeline access model that keeps the actual data separated. But it's not really possible because of this exact thing - we have to be able to say "stop here and wait for everything to finish executing" which wouldn't be possible in a single-thread JS-task-style model.)

Anyway, this should come up again when we try to set the correct timelines on all properties/slots. (I don't think we currently show any kind of warning indicator or console log when we see an algorithm or property used from the wrong timeline, but I bet we could.)

Reply:
> (I had a brief thought that maybe we could pretend all our stuff runs on the main thread too, which would be more web-spec-like, now that we have this nice timeline access model that keeps the actual data separated. But it's not really possible because of this exact thing - we have to be able to say "stop here and wait for everything to finish executing" which wouldn't be possible in a single-thread JS-task-style model.)

Reading this note in the spec, I realize perhaps this isn't true:

> Implementing WebGPU does not require timelines to execute in parallel, so does not require multiple processes, or even multiple threads.

We _probably could_ define things such that everything happens immediately (except things which post tasks like mapAsync and onSubmittedWorkDone, which would post the task at an unspecified point in the future).

Or we could define our timelines using ["parallelism"](https://html.spec.whatwg.org/multipage/infrastructure.html#parallelism) and rewrite that note to say that everything we specify as happening in parallel _could_ be done without any parallelism.